astrometry/src/PixelMap.cpp

```diff
@@ -350,3 +350,29 @@ ConstantMap::create(const YAML::Node& node,
   }
 }
 #endif
+
+PixelMap*
+ASTMap::duplicate() const {
+  return new ASTMap(*mapping);
```

Comment:
```suggestion
  return new ASTMap(mapping->copy());
```
We want this to be a real deep copy, not a shallow copy of just the `shared_ptr`.

Reply:
Two things--first isn’t `mapping` getting copied twice this way, because it gets copied again in the constructor? Second, should the constructor also be changed for this? As suggested, it doesn't compile 
(`no known conversion for argument 1 from 'std::shared_ptr<ast::Mapping>' to 'const ast::Mapping&'`).

Reply:
You're right - the copy is already there in the constructor, and we don't need another copy here.  Ignore me.